sbi/neural_nets/estimators/base.py

```diff
@@ -443,6 +509,447 @@ def diffusion_fn(self, input: Tensor, times: Tensor) -> Tensor:
+        raise NotImplementedError("Score is not implemented for this estimator.")
+
+    def mean_t_fn(self, times: Tensor) -> Tensor:
+        r"""Linear coefficient mean_t of the perturbation kernel expectation
+        :math:`\mu_t(t) = E[\theta_t | \theta_0] = \text{mean_t}(t) \cdot \theta_0`
+        specifying the "mean factor" at a given time, which is always multiplied by
+        :math:`\theta_0` to get the mean of the noise distribution, i.e.,
+        :math:`p(\theta_t | \theta_0) = N(\theta_t;
+                \text{mean_t}(t)*\theta_0, \text{std_t}(t)).`
+
+        Args:
+            times: SDE time variable in [0,1].
+
+        Raises:
+            NotImplementedError: Mean_t is not implemented for this estimator.
+        """
+        raise NotImplementedError("Mean_t is not implemented for this estimator.")
+
+    def std_fn(self, times: Tensor) -> Tensor:
+        r"""Standard deviation function std_t(t) of the perturbation kernel at a given
+            time,
+
+        .. math::
+            p(\theta_t | \theta_0) = N(\theta_t; \text{mean_t}(t) \cdot
+            \theta_0, \text{std_t}(t)^2).
+
+        Args:
+            times: SDE time variable in [0,1].
+
+        Raises:
+            NotImplementedError: Std_t is not implemented for this estimator.
+        """
+        raise NotImplementedError("Std_t is not implemented for this estimator.")
+
+    def drift_fn(self, input: Tensor, times: Tensor) -> Tensor:
+        r"""Drift function :math:`f(t)` of the vector field estimator.
+
+        The drift function :math:`f(t)` and diffusion function :math:`\g(t)`
+        enable SDE sampling:
+
+        .. math::
+            d\theta_t = [f(t) - g(t)^2 \nabla_{\theta_t} \log p(\theta_t | x_o)]dt
+              + \g(t) dW_t
+
+        where :math:`dW_t` is the Wiener process.
+
+
+        Args:
+            input: input parameters :math:`\theta_t`.
+            times: SDE time variable in [0,1].
+
+        Raises:
+            NotImplementedError: Drift is not implemented for this estimator.
+
+        """
+        raise NotImplementedError("Drift is not implemented for this estimator.")
+
+    def diffusion_fn(self, input: Tensor, times: Tensor) -> Tensor:
+        r"""Diffusion function :math:`\g(t)` of the vector field estimator.
+
+        The drift function :math:`f(t)` and diffusion function :math:`\g(t)`
+        enable SDE sampling:
+
+        .. math::
+            d\theta_t = [f(t) - g(t)^2 \nabla_{\theta_t} \log p(\theta_t | x_o)]dt
+              + \g(t) dW_t
+
+        where :math:`dW_t` is the Wiener process.
+
+        Args:
+            input: input parameters :math:`\theta_t`.
+            times: SDE time variable in [0,1].
+
+        Raises:
+            NotImplementedError: Diffusion is not implemented for this estimator.
+        """
+        raise NotImplementedError("Diffusion is not implemented for this estimator.")
+
+
+class MaskedConditionalVectorFieldEstimatorWrapper(ConditionalVectorFieldEstimator):
```

Comment:
Add docstring (especially on the beginning) what the purpuse of this class is and what it does.